ee/tabby-db/migrations/0022_github-provider.up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE github_repository_provider(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    display_name TEXT NOT NULL,
+    application_id TEXT NOT NULL,
+    secret TEXT NOT NULL,
+    CONSTRAINT `idx_application_id` UNIQUE (`application_id`)
+);
+CREATE TABLE github_provided_repositories(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    github_repository_provider_id INTEGER NOT NULL,
+    candidate_id TEXT NOT NULL,
+    name TEXT NOT NULL,
+    git_url TEXT NOT NULL,
+    CONSTRAINT `idx_provider_id_candidate_id` UNIQUE (`candidate_id`, `github_repository_provider_id`),
```

Comment:
```suggestion
    CONSTRAINT `idx_vendor_id` UNIQUE (`vendor_id`),
```

candidate_id is essentially github api vendor id (`node_id` in screenshot), which is a global unique id. 

<img width="485" alt="image" src="https://github.com/TabbyML/tabby/assets/388154/c75c23f4-d1fe-42ec-9140-0c7ee78a9b0a">